src/main/java/org/motechproject/mots/repository/custom/CommunityHealthWorkerRepositoryCustom.java

```diff
@@ -0,0 +1,13 @@
+package org.motechproject.mots.repository.custom;
+
+import org.motechproject.mots.domain.CommunityHealthWorker;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+public interface CommunityHealthWorkerRepositoryCustom {
+
+  Page<CommunityHealthWorker> searchCommunityHealthWorkers(
```

Comment:
Isn't is possible to genericize this custom so it takes a map of parameters? I suppose the search would work similarly in all our repositories

Reply:
@jkondrat 
I don't have idea how create simple generic repository with search and map of parameters.
If we get map of parameters, how will we know that the parameter exists in the entity? We need add validate that this fields exist in entity.
You get parameters as string. When we gets map of parameters, how to recognize what type the parameter should be? 
You gets strings in map, you don't know which type ths string represents in entity, for example it can be UUID, enum etc.
This means generate a possibility to create a lot of additional errors, the need to check and validate them before start searching.